services/dataplatform/cluster.go

```diff
@@ -129,9 +132,48 @@ func setPatchClusterRequestProperties(d *schema.ResourceData) *dataplatform.Patc
 		dataplatformCluster.Properties.MaintenanceWindow = setMaintenanceWindowData(d)
 	}
 
+	dataplatformCluster.Properties.Lans = setLansData(d)
+
 	return &dataplatformCluster
 }
 
+func setLansData(d *schema.ResourceData) *[]dataplatform.Lan {
+	var lansBody []dataplatform.Lan
+	if lansData, ok := d.GetOk("lans"); ok {
```

Comment:
maybe we can check != ok here and below and just return early so we don't have so much nesting?

Reply:
@cristiGuranIonos the code will look like this:
```
        var lansBody []dataplatform.Lan
	lansData, ok := d.GetOk("lans")
	if !ok {
		return lansBody
	}
	lansDataSet, ok := lansData.(*schema.Set)
	if !ok {
		return lansBody
	}
	for _, lanData := range lansDataSet.List() {
		if lan, ok := lanData.(map[string]interface{}); ok {
			lanID := lan["lan_id"].(string)
			dhcp := lan["dhcp"].(bool)
			var lanBody dataplatform.Lan
			lanBody.LanId = &lanID
			lanBody.Dhcp = &dhcp
			lanBody.Routes = setRoutesData(lan)
			lansBody = append(lansBody, lanBody)
		}
	}
	return lansBody
```
I prefer the first one with the nesting, I think it looks cleaner, what do you think?